client/homebrew/brewRenderer/headerNav/headerNav.jsx

```diff
@@ -0,0 +1,76 @@
+require('./headerNav.less');
+
+import * as React from 'react';
+import * as _ from 'lodash';
+
+
+const HeaderNav = React.forwardRef(({}, pagesRef)=>{
+
+	const [state, setState] = React.useState({
+		showHeaderNav : false
+	});
+
+	const toggleHeaderNav = ()=>{
+		setState((prevState)=>({
+			...prevState,
+			showHeaderNav : !prevState.showHeaderNav
+		}));
+	};
+
+	const renderHeaderLinks = ()=>{
+		if(!pagesRef.current) return;
+		const elements = pagesRef.current.querySelectorAll('[id]');
```

Comment:
Right, we can let CSS do the heavy lifting here:

```suggestion
		const elements = pagesRef.current.querySelectorAll('*:is(h1,h2,h3,h4,h5,h6)[id]');
```

We could even use this to leverage the depth level later.

Reply:
This should fix the aforementioned issue.

Reply:
But `id` can exist on more than just headers.

Reply:
Right, i thought we only wanted headings, but.. right, so the issue here is that any block with id gets caught, with all contents too, including in this case the table

Reply:
I've improved the text handling, which should prevent the entire content of a table with an ID being inserted into the header navigation.